make the remove icon of the file preview configurable and default to RemoveCircle (#8756)
marmelab/react-admin: docs/FileInput.md

|------------------------|----------|---------------------|-----------|---------------------------------------------------------------------|
| `accept`               | Optional | `string | string[]` | -          | Accepted file type(s). When empty, all file types are accepted.     |
| `children`             | Optional | `ReactNode`         | -         | Element used to preview file(s)                                     |
| `minSize`              | Optional | `number`            | 0         | Minimum file size (in bytes), e.g. 5000 for 5KB                     |
| `maxSize`              | Optional | `number`            | `Infinity` | Maximum file size (in bytes), e.g. 5000000 for 5MB                  |
| `multiple`             | Optional | `boolean`           | `false`   | Whether the inputs can accept multiple files.                       |
| `options`              | Optional | `Object`            | `{}`      | Additional options passed to react-dropzone's `useDropzone()` hook. |
| `placeholder`          | Optional | `ReactNode`         | -         | Invite displayed in the drop zone                                   |

> | `removeIcon`           | Optional | `ReactNode`         | [MUI's RemoveCircle icon](https://mui.com/material-ui/material-icons/?query=removeCir&selected=RemoveCircle) | The clickable icon for removing images                              |

Reviewer: ```suggestion
| `removeIcon`           | Optional | `ReactNode`         | [MUI's RemoveCircle icon](https://mui.com/material-ui/material-icons/?query=removeCir&selected=RemoveCircle) | The clickable icon for removing files                              |
```

| `multiple`             | Optional | `boolean`           | `false`    | Whether the inputs can accept multiple files.                       |
| `options`              | Optional | `Object`            | `{}`       | Additional options passed to react-dropzone's `useDropzone()` hook. |
| `placeholder`          | Optional | `ReactNode`         | -          | Invite displayed in the drop zone                                   |

> | `removeIcon`           | Optional | `ReactNode`         | -          | The clickable icon for removing images                              |

Reviewer: ```suggestion
| `removeIcon`           | Optional | `ReactNode`         | [MUI's RemoveCircle icon](https://mui.com/material-ui/material-icons/?query=removeCir&selected=RemoveCircle) | The clickable icon for removing files                              |
```

</FileInput>
```

## `removeIcon`

> Optionally overwrite the [default icon](https://mui.com/material-ui/material-icons/?query=removeCir&selected=RemoveCircle) for removing files.

Reviewer: Could you add a small code example showing how one can use `removeIcon` to change the default icon to something else? (take inspiration from the other sections of the doc :wink:)

Author: Done, and I tried to keep it consistent with the other sections.

</FileInput>
```

## `removeIcon`

> Optionally overwrite the [default icon](https://mui.com/material-ui/material-icons/?query=removeCir&selected=RemoveCircle) for removing files:

Reviewer: ```suggestion
Use the `removeIcon` prop to change the icon displayed as the remove button:
```

```jsx

> <ImageInput source="attachments" RemoveIcon={CustomSvgIcon}>
>     <ImageField source="src" title="title" />
> </ImageInput>

Reviewer: ```suggestion
<FileInput source="attachments" removeIcon={CustomSvgIcon}>
    <FileField source="src" title="title" />
</FileInput>
```
